FogCreek/Glitch-Community
# hazel-judge: Load Wistia scripts early, so landing page videos load faster

## Links
* [hazel-judge.glitch.me/create](https://hazel-judge.glitch.me/create)
* [#122](https://github.com/FogCreek/Glitch-Community-Work/issues/122) 

## Changes:
**Load Wistia scripts before our bundle and before analytics**
Once our app is done loading/parsing and renders `<WistiaVideo>` components, they are playable ASAP

This means that when we add a Wistia video, we'll need to specify the video's id in the route on the server like this:

```js
// routes.js
await render(res, { title, socialTitle, description, image, wistiaVideoId: '2vcr60pnx9'});
```
  
**Lazy load the embeds on /create**
Even though the Wistia scripts were being _loaded_ earlier, they weren't being _parsed_ until after the embeds and the embeds' scripts were finished. Lazy loading these embeds makes /create feel so much snappier!

**Update render function in routes.js to take an `opts` param instead of named paramaters**
This was to accommodate the growing list of parameters. Was becoming unwieldy and weird to specify each of them. 

## How To Test:
- Visit [hazel-judge.glitch.me/create](https://hazel-judge.glitch.me/create), it should feel fast! You should be able to play the Wistia video!
- Visit the [homepage](https://hazel-judge.glitch.me/), it should work and you should be able to play the Wistia video
- Click around a bunch of different page types. I updated how our render function in routes.js takes arguments, so we should confirm I didn't break anything. I already tested a ton of page types in metatags.io, so we should be good.
